packages/expo-battery/src/ExpoBattery.web.ts

```diff
@@ -70,18 +70,37 @@ export default {
   },
 };
 
-function onChargingChange(this: BatteryManager): void {
-  const batteryState = this.charging ? BatteryState.CHARGING : BatteryState.UNPLUGGED;
+let lastReportedState: BatteryState = BatteryState.UNKNOWN;
+
+function getBatteryState(isCharging: boolean, level: number) {
+  return isCharging
+    ? level >= 1.0
+      ? BatteryState.FULL
+      : BatteryState.CHARGING
+    : BatteryState.UNPLUGGED;
+}
+
+function emitStateChange(isCharging: boolean, level: number) {
+  const batteryState = getBatteryState(isCharging, level);
+  // prevent sending the same state change twice.
+  if (batteryState === lastReportedState) return;
   emitter.emit('Expo.batteryStateDidChange', { batteryState });
+  lastReportedState = batteryState;
+}
+
+function onChargingChange(this: BatteryManager): void {
+  emitStateChange(this.charging, this.level);
 }
 
 function onLevelChange(this: BatteryManager): void {
   const batteryLevel = this.level;
+  // update the state as well in case the state changed to full.
+  emitStateChange(this.charging, this.level);
   emitter.emit('Expo.batteryLevelDidChange', { batteryLevel });
 }
 
 async function getBatteryManagerAsync(): Promise<BatteryManager | null> {
-  if (!Platform.isDOMAvailable) return null;
+  if (Platform.isDOMAvailable === false) return null;
```

Comment:
This is just in case someone accidentally mismatches versions or if we publish it before unimodules/react-native-adapter is published. It's a pretty minor change which handles a lot more cases and in the future it can be removed.